packages/admin/cms-admin/src/generator/generateForm.ts

```diff
@@ -0,0 +1,358 @@
+                await refetch();
+            },
+        });
+    
+        const handleSubmit = async (state: FormValues, form: FormApi<FormValues>, event: FinalFormSubmitEvent) => {
+            if (await saveConflict.checkForConflicts()) {
+                throw new Error("Conflicts detected");
+            }
+    
+            const output = {
+                ...state,
+                ${numberFields.map((field) => `${field.name}: parseFloat(state.${field.name}),`).join("\n")}
+                ${Object.keys(rootBlocks)
+                    .map((rootBlockKey) => `${rootBlockKey}: rootBlocks.${rootBlockKey}.state2Output(state.${rootBlockKey}),`)
+                    .join("\n")}
+            };
+    
+            if (mode === "edit") {
+                if (!id) {
+                    throw new Error("Missing id in edit mode");
+                }
+                await client.mutate<GQLUpdate${entityName}Mutation, GQLUpdate${entityName}MutationVariables>({
+                    mutation: update${entityName}Mutation,
+                    variables: { id, input: output, lastUpdatedAt: data?.${instanceEntityName}?.updatedAt },
+                });
+            } else {
+                const { data: mutationReponse } = await client.mutate<GQLCreate${entityName}Mutation, GQLCreate${entityName}MutationVariables>({
+                    mutation: create${entityName}Mutation,
+                    variables: { ${hasScope ? `scope, ` : ""}input: output },
+                });
+                if (!event.navigatingBack) {
+                    const id = mutationReponse?.create${entityName}.id;
+                    if (id) {
+                        setTimeout(() => {
+                            stackSwitchApi.activatePage("edit", id);
+                        });
+                    }
+                }
+            }
+        };
+    
+        if (error) throw error;
+    
+        if (loading) {
+            return <CircularProgress />;
+        }
+    
+        return (
+            <FinalForm<FormValues>
+                apiRef={formApiRef}
+                onSubmit={handleSubmit}
+                mode={mode}
+                initialValues={initialValues}
+                onAfterSubmit={(values, form) => {
+                    //don't go back automatically
+                }}
+            >
+                {({ values }) => (
+                    <EditPageLayout>
+                        {saveConflict.dialogs}
+                        <Toolbar>
+                            <ToolbarItem>
+                                <IconButton onClick={stackApi?.goBack}>
+                                    <ArrowLeft />
+                                </IconButton>
+                            </ToolbarItem>
+                            <ToolbarTitleItem>
+                                {mode == "edit" ? values.title : <FormattedMessage id="${instanceNamePlural}.new${classNameSingular}" defaultMessage="New ${camelCaseToHumanReadable(
```

Comment:
`values.title` shouldn't be hardcoded. It's not certain that every entity has a title field. 

This should either be "Edit ${className}" or it must be configurable via the `crud-generator-config.ts`

Reply:
oops

I changed it to className for edit and add